### PR TITLE
Added Viewport Texture Tutorial

### PR DESCRIPTION
**Pull request by clayjohn**
This tutorial explains how to use a Viewport to generate a texture that can be applied to a sphere. In order to do so it walks the reader through the process of making a basic procedural planet.

In the future I would like to follow this tutorial up with two others. One showing how to create animated normal maps with shaders. And another showing how to use shaders to create an atmosphere effect for planets.

Please let me know if anything can be improved! I am very open to comments and constructive criticism.

### CODE REVIEW

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 1]
```diff
@@ -0,0 +1,321 @@
+.. _doc_viewport_as_texture:
+
+Using a Viewport as a texture
+=====================================
```
Comment (YeldhamDev) [inline tutorials/viewports/using_viewport_as_texture.rst]: You might want to remove the extraneous `=`'s.

**Comment by clayjohn** [comment]
Good catch. Thanks!

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+Introduction
+------------
+
+This tutorial will introduce you to using the :ref:`Viewport <class_Viewport>` as a
+texture that can be applied to 3D objects. In order to do so it will walk you through the process
+of making a procedural planet like the one below:
+
+.. image:: img/planet_example.png
+
+.. note:: This tutorial does not cover how to code a dynamic atmosphere like the one this planet has
```
Comment (cbscribe) [inline tutorials/viewports/using_viewport_as_texture.rst]: Missing a . at the end of the sentence.

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+Then from the menu that pops up select the Viewport that we rendered to earlier.
+
+.. image:: img/planet_pick_viewport_texture.png
+
+Your sphere should now be colored in with the colors we rendered to the Viewport
+
+.. image:: img/planet_seam.png
+
+Notice the ugly seem that forms where the texture texture wraps around? This is because we are picking
```
Comment (cbscribe) [inline tutorials/viewports/using_viewport_as_texture.rst]: seem -> seam

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+.. note:: If you are interested in a little extra information on the technique, we will be converting from
+          spherical coordinates into Cartesian coordinates. Spherical coordinates map the longitude and
+          latitude of the sphere, while Cartesian coordinates are for all intents and purposes a 
+          vector from the center of the sphere to the point. 
+
+For each pixel we will calculate its 3D position on the sphere. From that we will use
+3D noise to determine a color value. By calculating the noise in 3D we solve the problem
+of the pinching at the poles. To understand why, picture the noise being calculated across the 
+surface of the sphere instead of across the 2D plane. When you calculate across the 
+surface of the sphere you never hit an edge, and hence you never create a seem or
```
Comment (cbscribe) [inline tutorials/viewports/using_viewport_as_texture.rst]: seem -> seam

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+Now to use ``noised``, add the following to the  ``fragment`` function:
+
+::
+
+  vec4 n = noised(unit * 5.0);
+  COLOR.xyz = vec3(n.x * 0.5 + 0.5);
+
+.. image:: img/planet_noise.png
+
+.. note:: In order to highlight the texture, I have turned set the material to unshaded.
```
Comment (cbscribe) [inline tutorials/viewports/using_viewport_as_texture.rst]: remove "turned"

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+What ``smoothstep`` does is return ``0`` if the third parameter is below the first and return 1 if the
+third parameter is larger than the second and smoothly blends between ``0`` and ``1`` if the third number
+is between the first and the second. So in this line ``smoothstep`` returns ``0`` whenever ``n.x`` is less than ``-0.1``
+and it returns ``1`` whenever ``n.x`` is above ``0``.
+
+.. image:: img/planet_noise_smooth.png
+
+One more thing to make this a little more planet-y. The land shouldn't be so blobby lets make the edges
+a little rougher. A trick that is often used in shaders to make rough looking terrain with noise is
+to layer levels of noise over one another and various frequencies. We use one layer to make the
```
Comment (cbscribe) [inline tutorials/viewports/using_viewport_as_texture.rst]: I think "and" here should be a different word. "levels of noise over one another *using* various frequencies". "at" could work there as well.

File: tutorials/viewports/using_viewport_as_texture.rst [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+is between the first and the second. So in this line ``smoothstep`` returns ``0`` whenever ``n.x`` is less than ``-0.1``
+and it returns ``1`` whenever ``n.x`` is above ``0``.
+
+.. image:: img/planet_noise_smooth.png
+
+One more thing to make this a little more planet-y. The land shouldn't be so blobby lets make the edges
+a little rougher. A trick that is often used in shaders to make rough looking terrain with noise is
+to layer levels of noise over one another and various frequencies. We use one layer to make the 
+overall blobby structure of the continents. Then another layer breaks up the edges a bit, and then
+another and so on. What we will do is calculate ``n`` with four lines of shader code
```
Comment (cbscribe) [inline tutorials/viewports/using_viewport_as_texture.rst]: ", and so on."

**Comment by clayjohn** [comment]
Thanks @cbscribe! I addressed all your comments. 

I still can't believe I misspelled "seam" so many times. I guess thats what I get for writing this in vim...

**Comment by cbscribe** [comment]
I gave it some time in case others wanted to take a look. This is good content, so let's merge it!

**Pull request merged**